lib/internal/filter.js

```diff
@@ -1,31 +1,19 @@
-import arrayMap from 'lodash/_arrayMap';
-import property from 'lodash/_baseProperty';
+import filter from 'lodash/filter';
```

Comment:
the reason we've avoided importing `filter` up until this point is it included a bunch of deep property stuff (to support keys like `a.b['c']` that we do not need in async. I think I'd prefer we assume that `arr` is always an object and use `lodash` object methods instead of lodash collection methods